pandas/tests/series/test_constructors.py

```diff
@@ -773,11 +774,16 @@ def test_constructor_cast(self):
 
     def test_constructor_signed_int_overflow_raises(self):
         # GH#41734 disallow silent overflow, enforced in 2.0
-        msg = "Values are too large to be losslessly converted"
-        with pytest.raises(ValueError, match=msg):
+        if np_version_gt2:
+            msg = "The elements provided in the data cannot all be casted to the dtype"
+            err = OverflowError
```

Comment:
should we be catching and re-raising?

Reply:
Not sure if it's worth it. I like the new error from numpy better anyways.

Do we need to preserve backwards compatibility here?

Reply:
I am OK with this - I am not sure I see the value in re-raising as a ValueError when the OverflowError is more descriptive of what went wrong to begin with

Reply:
agreed OverflowError seems More Correct; my main opinion is that we should be consistent (no idea what the status quo is consistency-wise)

Reply:
Yea that's a good point. I wouldn't mind generally not replacing OverflowError with ValueError; they are both part of the stdlib but the former is more descriptive. 

I am unaware of the history behind why that pattern might have been implemented